frontend/src/views/RoleView.vue

```diff
@@ -97,6 +139,79 @@ const createKnowledge = async () => {
 
   knowledge.value.name = ''
 }
+
+const deleteDialogVisible = ref(false)
+const levelToDelete = ref(0)
+
+const openDeleteLevelButton = (row: Level) => {
+  levelToDelete.value = Number(row.id)
+  deleteDialogVisible.value = true
+}
+
+const closeDeleteDialog = () => {
+  deleteDialogVisible.value = false
+}
+
+async function deleteLevel() {
+  try {
+    await levelsStore.removeLevel(levelToDelete.value)
+    SuccessMessage('Level removed!')
+    closeDeleteDialog()
+  } catch (error) {
+    ErrorMessage('Failed to delete level')
+  }
+}
+
+const editKnowledgeFormVisible = ref(false)
+const knowledgeStore = useKnowledgeStore()
+const knowledgeToUpdate = ref<Knowledge>({
```

Comment:
Is this variable necessary? You can maybe put this inside your dialog component, also